// astro-decap-cms-oauth/dist/astro-decap-cms-oauth.js
const i = {
  adminDisabled: !1,
  adminRoute: "/admin",
  oauthDisabled: !1,
  oauthLoginRoute: "/oauth",
  oauthCallbackRoute: "/oauth/callback"
};
function c(n) {
  const { adminDisabled: r, adminRoute: t, oauthDisabled: e, oauthLoginRoute: a, oauthCallbackRoute: s } = {
    ...i,
    ...n
  };
  if (!(t != null && t.startsWith("/")) || !(a != null && a.startsWith("/")) || !(s != null && s.startsWith("/")))
    throw new Error('`adminRoute`, `oauthLoginRoute` and `oauthCallbackRoute` options must start with "/"');
  return {
    name: "astro-decap-cms-oauth",
    hooks: {
      "astro:config:setup": async ({ injectRoute: o }) => {
        r || o({
          pattern: t,
          entrypoint: "astro-decap-cms-oauth/src/admin.astro"
        }), e || (o({
          pattern: a,
          entrypoint: "astro-decap-cms-oauth/src/oauth/index.ts"
        }), o({
          pattern: s,
          entrypoint: "astro-decap-cms-oauth/src/oauth/callback.ts"
        }));
      }
    }
  };
}
export {
  c as default
};
